File: backend/services/PaymentService/PaymentService.API/Application/Dtos/PagedResult.cs
namespace PaymentService.API.Application.Dtos
{
    public class PagedResult<T>
    {
        public int Skip { get; private set; }
        public int Limit { get; private set; }
        public long Total { get; private set; }
        public bool End { get; private set; }
        public IEnumerable<T> Items { get; private set; }

        public PagedResult(IEnumerable<T> items, int skip, int limit, long total)
        {
            Items = items;
            Skip = skip;
            Limit = limit;
            Total = total;
            End = limit + skip >= total;
        }
    }
}

File: backend/services/user_service/src/utils/paged-result.ts
export default class PagedResult<T> {
    skip: number;
    limit: number;
    total: number;
    end: boolean;
    items: T[];

    constructor(items: T[], skip: number, limit: number, total: number) {
        this.skip = skip;
        this.limit = limit;
        this.total = total;
        this.end = limit + skip >= total;
        this.items = items;        
    }
}